client/v3/maintenance.go

```diff
@@ -57,10 +57,10 @@ type Maintenance interface {
 	// is non-zero, the hash is computed on all keys at or below the given revision.
 	HashKV(ctx context.Context, endpoint string, rev int64) (*HashKVResponse, error)
 
-	// Snapshot provides a reader for a point-in-time snapshot of etcd.
+	// Snapshot returns a reader for a point-in-time snapshot and version of etcd that created it.
 	// If the context "ctx" is canceled or timed out, reading from returned
 	// "io.ReadCloser" would error out (e.g. context.Canceled, context.DeadlineExceeded).
-	Snapshot(ctx context.Context) (io.ReadCloser, error)
+	Snapshot(ctx context.Context) (io.ReadCloser, string, error)
```

Comment:
client/v3 is considered public API, this is breaking change according to package compatiblility rules. 

Would consider having a Snapshot2(SnapshotInfo, error) method side by side with the expanded API
(and marking the other method as deprecated).

Reply:
Could we have the new method return `SnapshotReponse` and error? Similar to other function signatures in this package.

Nit suggestion: Maybe instead of `Snapshot2` it could be `SnapshotWtihResponse`.